template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Areas/IdentityService/Controllers/IdentityServiceController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity.Service;
+using Microsoft.AspNetCore.Identity.Service.IntegratedWebClient;
+using Microsoft.AspNetCore.Identity.Service.Mvc;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Company.WebApplication1.Identity.Models;
+
+namespace Company.WebApplication1.Identity.Controllers
+{
+    [Area(IdentityServiceConstants.Tenant)]
+    [Route("tfp/[area]/" + IdentityServiceConstants.DefaultPolicy + "/oauth2/v" + IdentityServiceConstants.Version + "/[action]")]
+    public class IdentityServiceController : Controller
+    {
+        private readonly IOptions<IdentityServiceOptions> _options;
+        private readonly ITokenManager _tokenManager;
+        private readonly SessionManager<ApplicationUser, IdentityServiceApplication> _sessionManager;
+        private readonly IAuthorizationResponseFactory _authorizationResponseFactory;
+        private readonly ITokenResponseFactory _tokenResponseFactory;
+
+        public IdentityServiceController(
+            IOptions<IdentityServiceOptions> options,
```

Comment:
Note: given that Identity itself uses IOptions to access its options, its probably not a good idea to use IOptionsSnapshot to read IdentityOptions in ClientInfoProvider since that only introduces a possibility that they get out of sync.